mesonbuild/compilers/c.py

```diff
@@ -744,6 +744,11 @@ def thread_link_flags(self):
         return ['-pthread']
 
     def has_multi_arguments(self, args, env):
+        # some compilers, e.g. GCC, don't warn for unsupported warning-disable
+        # flags, so when we are testing a flag like "-Wno-forgotten-towel", also
+        # check the equivalent enable flag too "-Wforgotten-towel"
+        if len(args) == 1 and args[0].startswith('-Wno-'):
+                args.append(args[0].replace('-Wno-', '-W'))
```

Comment:
`replace` goes over the whole string. This would be better written as `'-W' + args[0][5:]`.